tests/integration/synapseclient/models/async/test_schema_organization.py

```diff
@@ -0,0 +1,397 @@
+    js2 = JSONSchema("schema2", name)
+    js3 = JSONSchema("schema3", name)
+    js1.store({})
+    js2.store({})
+    js3.store({})
+
+    def delete_org():
+        for schema in org.get_json_schemas():
+            schema.delete()
+        org.delete()
+
+    request.addfinalizer(delete_org)
+
+    return org
+
+
+class TestSchemaOrganization:
+    """Asynchronous integration tests for SchemaOrganization."""
+
+    @pytest.fixture(autouse=True, scope="function")
+    def init(self, syn: Synapse) -> None:
+        self.syn = syn
+
+    @pytest.mark.asyncio
+    async def test_create_and_get(self, organization: SchemaOrganization) -> None:
+        # GIVEN an initialized organization object that hasn't been stored in Synapse
+        # THEN it shouldn't have any metadata besides it's name
+        assert organization.name is not None
+        assert organization.id is None
+        assert organization.created_by is None
+        assert organization.created_on is None
+        # AND it shouldn't exist in Synapse
+        assert not org_exists(organization.name, synapse_client=self.syn)
+        # WHEN I store the organization the metadata will be saved
+        await organization.store_async(synapse_client=self.syn)
+        assert organization.name is not None
+        assert organization.id is not None
+        assert organization.created_by is not None
+        assert organization.created_on is not None
+        # AND it should exist in Synapse
+        assert org_exists(organization.name, synapse_client=self.syn)
+        # AND it should be getable by future instances with the same name
+        org2 = SchemaOrganization(organization.name)
+        await org2.get_async(synapse_client=self.syn)
+        assert organization.name is not None
+        assert organization.id is not None
+        assert organization.created_by is not None
+        assert organization.created_on is not None
+        # WHEN I try to store an organization that exists in Synapse
+        # THEN I should get an exception
+        with pytest.raises(SynapseHTTPError):
+            org2.store()
+
+    @pytest.mark.asyncio
+    async def test_get_json_schemas_async(
+        self,
+        organization: SchemaOrganization,
+        organization_with_schema: SchemaOrganization,
+    ) -> None:
+        # GIVEN an organization with no schemas and one with 3 schemas
+        await organization.store_async(synapse_client=self.syn)
+        # THEN get_json_schema_list should return the correct list of schemas
+        schema_list = []
+        async for item in organization.get_json_schemas_async(synapse_client=self.syn):
+            schema_list.append(item)
+        assert len(schema_list) == 0
+        schema_list2 = []
+        async for item in organization_with_schema.get_json_schemas_async(
+            synapse_client=self.syn
+        ):
+            schema_list2.append(item)
+        assert len(schema_list2) == 3
+
+    @pytest.mark.asyncio
```

Comment:
Don't use this mark on any tests. It is not needed, in the pytest.ini it's set to automatic mode